### ISSUE TITLE
tartufo entropy scanning does not handle base64url encoding

### ISSUE THREAD

**Issue by pmevzek-godaddy** [godaddy/tartufo#177]
## 🐛 Bug Report

There are multiple flavors (distinct in the alphabet used) of base64.

Tartufo source code at `tartufo/scanner.py` has
```
BASE64_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
```

This doesn't match for example the base 64 "url" flavor that uses `_` and `-` (instead of `+`. and `/`) characters as defined in §5 of RFC4648.

## To Reproduce

```
$ git init test
$ cd test
$ git commit --allow-empty --allow-empty-message -m 'Start'
[master (root-commit) a64409a] Start
$ echo -e "g34\xFFahChi\xFAk91PWpbG" | base64 > secret_base64.txt
$ echo -e "g34\xFFahChi\xFAk91PWpbG" | base64 | tr + - | tr / _ > secret_base64url.txt
$ cat secret_base64*
ZzM0/2FoQ2hp+ms5MVBXcGJHCg==
ZzM0_2FoQ2hp-ms5MVBXcGJHCg==
$ git add secret_base64*
$ tartufo --entropy --regex pre-commit
~~~~~~~~~~~~~~~~~~~~~
Reason: High Entropy
Filepath: secret_base64.txt
Signature: 56d5080cddeeb2c434550afd809c43c06bef51f799b96cdc697f436e5270350e
@@ -0,0 +1 @@
+ZzM0/2FoQ2hp+ms5MVBXcGJHCg==

~~~~~~~~~~~~~~~~~~~~~
```

## Expected Behavior

All flavors of base64 should be treated the same concerning entropy, so in above example both files should be reported.

## Code Example

N/A

## Environment

tartufo version 2.4.0

**Comment by pmevzek-godaddy** [comment]
FWIW, JWK standard (RFC 7517) aka JSON Web Key or a way to encode public/private keys in a JSON structure with needed metadata explicitly uses base64url variant almost everywhere (and not base64) to encode content, hence the high entropy scanner might miss it (and otherwise there would be a need to have a "JSON scanner" able to detect structures that are JWK/JWE/JWS things and flag them). The related standards (JWS and JWE) are in the same boat, all for obvious reasons, as they are tailored for web operations.

Here is one example from the specification:

>    Assemble the final representation: The JWE Compact Serialization of
>    this result, as defined in Section 7.1 of [JWE], is the string
>    BASE64URL(UTF8(JWE Protected Header)) || '.' || BASE64URL(JWE
>    Encrypted Key) || '.' || BASE64URL(JWE Initialization Vector) || '.'
>    || BASE64URL(JWE Ciphertext) || '.' || BASE64URL(JWE Authentication
>    Tag).

**Comment by rbailey-godaddy** [comment]
On review, [RFC4648](https://datatracker.ietf.org/doc/html/rfc4648) is pretty explicit that there is only one "base64", and that's the one tartufo implements. I have recategorized this as a feature request rather than bug report, with the explicit intention of adding support for "base64url" (i.e. the alternative representation described in section 5).

**Comment by tarkatronic** [comment]
Fixed in #274
